server/services/v1/sessions.go

```diff
@@ -45,6 +56,64 @@ type SessionManager struct {
 	txListeners   []TxListener
 }
 
+type SessionManagerWithMetrics struct {
+	s *SessionManager
+}
+
+func (m *SessionManagerWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
+	tags := metrics.GetSessionTags(ctx, name)
+	spanMeta := metrics.NewSpanMeta(metrics.SessionManagerServiceName, name, "session", tags)
+	defer metrics.SessionRespTime.Tagged(spanMeta.GetTags()).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
+	ctx = spanMeta.StartTracing(ctx, true)
+	if err := f(ctx); err != nil {
+		spanMeta.CountErrorForScope(metrics.SessionErrorRequests, err)
+		spanMeta.FinishWithError(ctx, err)
+		return
+	}
+	spanMeta.CountOkForScope(metrics.SessionOkRequests)
+	_ = spanMeta.FinishTracing(ctx)
+}
+
+func (m SessionManagerWithMetrics) Create(ctx context.Context, trackVerInOwnTxn bool, instantVerTracking bool, track bool) (qs *QuerySession, err error) {
+	m.measure(ctx, "Create", func(ctx context.Context) error {
+		qs, err = m.s.Create(ctx, trackVerInOwnTxn, instantVerTracking, track)
+		return err
+	})
+	return
+}
+
+func (m SessionManagerWithMetrics) Get(ctx context.Context) (qs *QuerySession, err error) {
```

Comment:
We can skip measuring Get/Remove because this is an in-memory operation only so overhead here will be mostly for reporting metrics.